src/views/Transitioner.js

```diff
@@ -155,7 +155,11 @@ class Transitioner extends React.Component<*, Props, State> {
     const { timing } = transitionSpec;
     delete transitionSpec.timing;
 
-    const animations = indexHasChanged
+    const toValue = nextProps.navigation.state.index;
+    const positionHasChanged = position.__getValue() !== toValue;
```

Comment:
but actually, the former animation has already completed, and the following animation is on the go, so I think it's safe to do it this way https://github.com/react-community/react-navigation/blob/master/src/views/CardStack.js#L208
BTM, mind to review #1110 now?